### ISSUE TITLE
Suggested Feature in data.gov search: Autofill

### ISSUE THREAD

**Issue by lunawat** [GSA/datagov-wptheme#576]
Is there a reason why the search feature in data.gov does not have autofill like Google?

**Comment by philipashlock** [comment]
@kvuppala Do you know if this is possible with ["autocomplete" in the API](http://docs.ckan.org/en/latest/api/index.html#ckan.logic.action.get.package_autocomplete) or if that's even used anywhere?

I don't think I've seen any CKAN instances where the main dataset searchbox has autocomplete.

**Comment by kvuppala** [comment]
Yes seems like package_autocomplete provides the datasets titles with the keywords typed in,  not seen on any CKAN instances, but it must be using the underlying Solr capabilities, which we had used in other sites before.

**Comment by JJediny** [comment]
This would require use to build in a cache of titles from Solr/API as performance could be severely affected if done as a stand-alone request each time.  Plugins used for reporting and harvesting do already use a asynchronous task manager in [celery](http://www.celeryproject.org/) that could be extended to include a task to create one. Just a matter of priorities of current feature-requests
